sdk/daml-lf/ledger-api-value/src/main/protobuf/com/daml/ledger/api/v2/value.proto

```diff
@@ -59,97 +75,80 @@ message Value {
     string text = 8;
 
     // Identifier of an on-ledger contract. Commands which reference an unknown or already archived contract ID will fail.
-    // Must be a valid LedgerString.
-    string contract_id = 9;
+    // Must encode a valid LedgerString.
+    bytes contract_id = 9;
```

Comment:
This is the most significant change in this PR. The contract-id changes type to `bytes`

Reply:
@remyhaemmerle-da 
I get a lot of questions about contract_id being bytes. I thought there was some deep reason for that related to canton protocol. But there doesn't seem to be one. Could we go in the opposite direction and redefine daml-lf's contract-id as string?

Reply:
It used to be a ContractId message that contained a string. It became bytes  months ago:
https://github.com/digital-asset/daml/commit/8227412a6bb2ac4bf68a481ead81696154e1a0bf

Reply:
There is no deep reason about represented contract id as string expect 
- using a closer reprensantation of the engine internal
- saving space (as string they need 140bytes as bytestring we need 70bytes)

> It used to be a ContractId message that contained a string. It became bytes 6 months ago:

Reply:
If we want to revert back, we will need to either 
- add a new field for new LF version
- keep the bytes type and populate it with string.